### PR TITLE
[Rules] Add Skills:TrivialTradeskillCombinesNoFail Rule

### PR DESCRIPTION
**Pull request by JasXSL**
Adds a server rule that (when enabled) removes the fail chance for trivial tradeskill combinations.

### CODE REVIEW

File: zone/tradeskills.cpp [hunk 1]
```diff
@@ -1050,6 +1050,9 @@ bool Client::TradeskillExecute(DBTradeskillRecipe_Struct *spec) {
 		// above critical still stands.
 		// Mastery modifier is: 10%/25%/50% for rank one/two/three
 		chance = 95.0f + (float(user_skill - spec->trivial) / 40.0f);
+		if(RuleB(Skills, TrivialNoFail)) {
```
Comment (Kinglykrab) [inline zone/tradeskills.cpp:1053]: `if (` instead of `if(`

Comment (Kinglykrab) [reply zone/tradeskills.cpp:1053]: Some newlines around this would make it easier to read.

File: common/ruletypes.h [hunk 2]
```diff
@@ -241,6 +241,7 @@ RULE_CATEGORY_END()
 RULE_CATEGORY(Skills)
 RULE_INT(Skills, MaxTrainTradeskills, 21, "Highest level for trading skills that can be learnt by the trainer")
 RULE_BOOL(Skills, UseLimitTradeskillSearchSkillDiff, true, "Enables the limit for the maximum difference between trivial and skill for recipe searches and favorites")
+RULE_BOOL(Skills, TrivialNoFail, false, "Makes all trivial recipes no-fail")
```
Comment (Kinglykrab) [inline common/ruletypes.h:244]: `TrivialRecipesNoFail` is a bit more verbose and explains what this does better.

File: utils/sql/svn/1392_recipe_learning.sql [hunk 3]
```diff
@@ -9,4 +9,5 @@ alter table `tradeskill_recipe` add column `must_learn` tinyint not null default
 
 insert into rule_values (ruleset_id, rule_name, rule_value, notes) values
 (1, 'Skills:UseLimitTradeskillSearchSkillDiff', 'true', 'Enables the limit for the maximum difference between trivial and skill for recipe searches and favorites.'),
-(1, 'Skills:MaxTradeskillSearchSkillDiff', '50', 'The maximum difference in skill between the trivial of an item and the skill of the player if the trivial is higher than the skill. Recipes that have not been learnt or made at least once via the Experiment mode will be removed from searches based on this criteria.');
\ No newline at end of file
+(1, 'Skills:TrivialNoFail', 'false', 'Makes all trivial recipes no-fail.'),
```
Comment (Kinglykrab) [inline utils/sql/svn/1392_recipe_learning.sql:12]: Adding this here is unnecessary, rules are handled automatically by the source.

File: common/ruletypes.h (hunk 2 above)
Comment (JasXSL) [reply common/ruletypes.h:244]: May as well go more verbose with "TrivialTradeskillCombinesNoFail", which better matches the otehr two Skills rules

Comment (Kinglykrab) [reply common/ruletypes.h:244]: Works for me. :+1:

File: utils/sql/svn/1392_recipe_learning.sql (hunk 3 above)
Comment (JasXSL) [reply utils/sql/svn/1392_recipe_learning.sql:12]: So that SQL file is basically deprecated then?

Comment (Kinglykrab) [reply utils/sql/svn/1392_recipe_learning.sql:12]: Yes, 100%, it's from SVN (subversion) days which is over 10 years old at this point, long before this repository existed.

Comment (Kinglykrab) [reply utils/sql/svn/1392_recipe_learning.sql:12]: We automatically add new rules in `rulesys.cpp`.
```cpp
	for (const auto &d : rule_data) {
		if (std::find(v.begin(), v.end(), d.first) == v.end()) {
			injected_rule_entries.push_back(
				std::tuple<int, std::string, std::string, std::string>(
					rule_set_id,
					d.first,
					d.second.first,
					Strings::Escape(*d.second.second)
				)
			);

			if (!quiet_update) {
				LogInfo(
					"Adding new rule [{}] ruleset [{}] ({}) value [{}]",
					d.first,
					rule_set_name,
					rule_set_id,
					d.second.first
				);
			}
		}
	}
	```

**Review by Kinglykrab — APPROVED** [review]

**Pull request merged**